source/runtime.hpp

```diff
@@ -155,6 +163,9 @@ namespace reshade
 		/// Callback function called to apply the post-processing effects to the screen.
 		/// </summary>
 		void on_present_effect();
+		bool is_d3d9 = false;
+		bool is_d3d10 = false;
+		bool is_d3d11 = false;
```

Comment:
There is `_renderer_id` for that. These are not needed. Use the following in-place where needed instead:
```
const bool is_d3d9 = (_renderer_id & 0x9000) != 0;
const bool is_d3d10 = (_renderer_id & 0xa000) != 0;
const bool is_d3d11 = (_renderer_id & 0xb000) != 0;
```

Reply:
I know, but in my rig, it did not work... Will retest

Reply:
Have retested it, and still does not work on my rig. On the other hand, the fact of setting those bool vars in the init stage of the runtime work perfectly.

Reply:
Sorry, the code I wrote is wrong. It's actually this:
```c++
const bool is_d3d9 = _renderer_id < 0xa000;
const bool is_d3d10 = _renderer_id >= 0xa000 && _renderer_id < 0xb000;
const bool is_d3d11 = _renderer_id >= 0xb000 && _renderer_id < 0xc000;
```